File: Practice/src/exercise/RudimentaryKeyboard.java
package exercise;

import java.util.HashMap;
import java.util.Map;

public class RudimentaryKeyboard {
//	대충 만든 자판
	public int[] solution(String[] keymap, String[] targets) {
		int[] answer = new int[targets.length];
		Map<Character, Integer> m = new HashMap<Character, Integer>();

		for (int i = 0; i < 26; i++) {
			char s = (char) (i + 65);
			int min = 101;
			for (String k : keymap) {
				int idx = k.indexOf(s);
				if (idx > -1) {
					min = Math.min(min, idx);
				}
			}
			m.put(s, min > 100 ? -1 : min);
		}

		for (int i = 0; i < targets.length; i++) {
			int count = 0;
			for (int j = 0; j < targets[i].length(); j++) {
				int n = m.get(targets[i].charAt(j));
				if (n < 0) {
					count = -1;
					break;
				}
				count += n + 1;
			}
			answer[i] = count;
		}

		return answer;
	}
}

File: Practice/src/exercise/RudimentaryKeyboard.py
def solution(keymap, targets):
    answer = []
    tap = {}
    for keys in keymap:
        for i, k in enumerate(keys):
            tap[k] = min(tap.get(k, 101), i + 1)
    for target in targets:
        sum = 0
        for t in target:
            n = tap.get(t, -1)
            if n == -1:
                sum = n
                break
            sum += n
        answer.append(sum)
    return answer
